inference-engine/src/mkldnn_plugin/nodes/mkldnn_fullyconnected_node.cpp

```diff
@@ -17,20 +17,47 @@ using namespace mkldnn;
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
 
+bool MKLDNNFullyConnectedNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto fc = std::dynamic_pointer_cast<const FullyConnectedNode>(op);
+        if (!fc) {
+            errorMessage = "Only legacy FullyConnected operation is supported";
+            return false;
+        }
+        if (std::dynamic_pointer_cast<const ngraph::opset1::Constant>(fc->get_input_node_shared_ptr(WEIGHTS_ID)) == nullptr) {
+            errorMessage = "Only Constant operation on 'weights' input is supported";
+            return false;
+        }
+        if (std::dynamic_pointer_cast<const ngraph::opset1::Constant>(fc->get_input_node_shared_ptr(BIAS_ID)) == nullptr) {
+            errorMessage = "Only Constant operation on 'bias' input is supported";
+            return false;
+        }
+        if (fc->get_input_shape(DATA_ID).size() != 2) {
+            errorMessage = "Only 'data' input with rank = 2 is supported";
+            return false;
+        }
+        if (fc->get_input_shape(WEIGHTS_ID).size() != 2) {
+            errorMessage = "Only 'weights' input with rank = 2 is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
 MKLDNNFullyConnectedNode::MKLDNNFullyConnectedNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
-        : MKLDNNNode(op, eng, cache), withBiases(false), baseInputsNumber(0) {
-//    internalBlobDesc.emplace_back([&](primitive_desc_iterator &primitive_desc_it, size_t idx) -> MKLDNNMemoryDesc {
-//        return MKLDNNMemoryDesc(primitive_desc_it.weights_desc(0));
-//    });
-//    internalBlobDesc.emplace_back([&](primitive_desc_iterator &primitive_desc_it, size_t idx) -> MKLDNNMemoryDesc {
-//        if (internalBlobs.size() <= 1)
-//            return MKLDNNMemoryDesc();
-//        return MKLDNNMemoryDesc(primitive_desc_it.weights_desc(1));
-//    });
-//
-//    if (getCnnLayer()->type == "FullyConnected" || getCnnLayer()->type == "InnerProduct") {
-//        baseInputsNumber = getCnnLayer().get()->insData.size();
-//    }
+        : MKLDNNNode(op, eng, cache), withBiases(false) {
+    std::string errorMessage;
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "FullyConnected node with name '" + getName() + "'";
+
+        const auto bias = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(op->get_input_node_shared_ptr(BIAS_ID))->cast_vector<float>();
+        withBiases = std::any_of(bias.begin(), bias.end(), [](float b){ return b != 0.0f; });
```

Comment:
Cannot we just don't create bias input during transformation if all values are zero?

Reply:
transformations was changed